kernel/build.sbt

```diff
@@ -0,0 +1,201 @@
+ */
+
+// Looks some of release settings should be set for the root project as well.
+publishArtifact := false  // Don't release the root project
+publish / skip := true
+publishTo := Some("snapshots" at "https://oss.sonatype.org/content/repositories/snapshots")
+releaseCrossBuild := false
+releaseProcess := Seq[ReleaseStep](
+  checkSnapshotDependencies,
+  inquireVersions,
+  runTest,
+  setReleaseVersion,
+  commitReleaseVersion,
+  tagRelease,
+  releaseStepCommandAndRemaining("+publishSigned"),
+  setNextVersion,
+  commitNextVersion
+)
+
+/**
+ * Release settings for artifact that contains only Java source code
+ */
+lazy val javaOnlyReleaseSettings = Seq(
+  // drop off Scala suffix from artifact names
+  crossPaths := false,
+  // we publish jars for each scalaVersion in crossScalaVersions. however, we only need to publish
+  // one java jar. thus, only do so when the current scala version == default scala version
+  publishArtifact := scalaBinaryVersion.value == "2.12",
+  // exclude scala-library from dependencies in generated pom.xml
+  autoScalaLibrary := false,
+)
+
+lazy val releaseSettings = Seq(
+  publishMavenStyle := true,
+  publishArtifact := true,
+  Test / publishArtifact := false,
+  releasePublishArtifactsAction := PgpKeys.publishSigned.value,
+  releaseCrossBuild := true,
+  pgpPassphrase := sys.env.get("PGP_PASSPHRASE").map(_.toArray),
+  sonatypeProfileName := "io.delta", // sonatype account domain name prefix / group ID
+  credentials += Credentials(
+    "Sonatype Nexus Repository Manager",
+    "oss.sonatype.org",
+    sys.env.getOrElse("SONATYPE_USERNAME", ""),
+    sys.env.getOrElse("SONATYPE_PASSWORD", "")
+  ),
+  publishTo := {
+    val nexus = "https://oss.sonatype.org/"
+    if (isSnapshot.value) {
+      Some("snapshots" at nexus + "content/repositories/snapshots")
+    } else {
+      Some("releases"  at nexus + "service/local/staging/deploy/maven2")
+    }
+  },
+  licenses += ("Apache-2.0", url("http://www.apache.org/licenses/LICENSE-2.0")),
+  pomExtra :=
+    <url>https://github.com/delta-io/connectors</url>
+      <scm>
+        <url>git@github.com:delta-io/connectors.git</url>
+        <connection>scm:git:git@github.com:delta-io/connectors.git</connection>
+      </scm>,
+) ++ javaOnlyReleaseSettings
+
+/*
+ ********************
+ *  MIMA settings   *
+ ********************
+ */
+def getPrevVersion(currentVersion: String): String = {
+  implicit def extractInt(str: String): Int = {
+    """\d+""".r.findFirstIn(str).map(java.lang.Integer.parseInt).getOrElse {
+      throw new Exception(s"Could not extract version number from $str in $version")
+    }
+  }
+
+  val (major, minor, patch): (Int, Int, Int) = {
+    currentVersion.split("\\.").toList match {
+      case majorStr :: minorStr :: patchStr :: _ =>
+        (majorStr, minorStr, patchStr)
+      case _ => throw new Exception(s"Could not find previous version for $version.")
+    }
+  }
+
+  val majorToLastMinorVersions: Map[Int, Int] = Map(
```

Comment:
this complexity is not needed right now. lets remove the entire mima related stuff. we can easily add it when needed. lets keep it simple.